src/modules/Game.class.js

```diff
@@ -20,25 +16,256 @@ class Game {
   }
 
-  moveLeft() {}
-  moveRight() {}
-  moveUp() {}
-  moveDown() {}
+  moveLeft() {
+    if (this.status !== 'playing') {
+      return;
+    }
+
+    let moved = false;
+
+    for (let i = 0; i < 4; i++) {
+      const row = this.board[i].filter((val) => val !== 0);
+      const newRow = [];
+
+      for (let j = 0; j < row.length; j++) {
+        if (row[j] === row[j + 1]) {
+          newRow.push(row[j] * 2);
+          this.score += row[j] * 2;
+          j++;
+          moved = true;
+        } else {
+          newRow.push(row[j]);
+        }
+      }
+
+      while (newRow.length < 4) {
+        newRow.push(0);
+      }
+
+      if (this.board[i].toString() !== newRow.toString()) {
+        this.board[i] = newRow;
+        moved = true;
+      }
+    }
+
+    if (moved) {
+      this.addRandomTile();
+    }
+
+    this.checkGameOver();
+  }
+
+  moveRight() {
+    if (this.status !== 'playing') {
+      return;
+    }
+
+    let moved = false;
+
+    for (let i = 0; i < 4; i++) {
+      const row = this.board[i].filter((val) => val !== 0).reverse();
+      const newRow = [];
+      let j = 0;
+
+      while (j < row.length) {
+        if (row[j] === row[j + 1]) {
+          newRow.push(row[j] * 2);
+          this.score += row[j] * 2;
+          j += 2;
+          moved = true;
+        } else {
+          newRow.push(row[j]);
+          j++;
+        }
+      }
+
+      while (newRow.length < 4) {
+        newRow.push(0);
+      }
+
+      newRow.reverse();
+
+      if (this.board[i].toString() !== newRow.toString()) {
+        this.board[i] = newRow;
+        moved = true;
+      }
+    }
+
+    if (moved) {
+      this.addRandomTile();
+    }
+
+    this.checkGameOver();
+  }
+
+  moveUp() {
+    if (this.status !== 'playing') {
+      return;
+    }
+
+    let moved = false;
+
+    for (let col = 0; col < 4; col++) {
+      const column = [];
+
+      for (let row = 0; row < 4; row++) {
+        if (this.board[row][col] !== 0) {
+          column.push(this.board[row][col]);
+        }
+      }
+
+      const newColumn = [];
+      let i = 0;
+
+      while (i < column.length) {
+        if (column[i] === column[i + 1]) {
+          newColumn.push(column[i] * 2);
+          this.score += column[i] * 2;
+          i += 2;
+        } else {
+          newColumn.push(column[i]);
+          i++;
+        }
+      }
+
+      while (newColumn.length < 4) {
+        newColumn.push(0);
+      }
+
+      for (let row = 0; row < 4; row++) {
+        if (this.board[row][col] !== newColumn[row]) {
+          moved = true;
+        }
+        this.board[row][col] = newColumn[row];
+      }
+    }
+
+    if (moved) {
+      this.addRandomTile();
+    }
+
+    this.checkGameOver();
+  }
+
+  moveDown() {
+    if (this.status !== 'playing') {
+      return;
+    }
+
+    let moved = false;
+
+    for (let col = 0; col < 4; col++) {
+      const column = [];
+
+      for (let row = 0; row < 4; row++) {
+        if (this.board[row][col] !== 0) {
```

Comment:
The `moveDown` method uses `this.board`. Ensure that `this.board` is initialized correctly before using it in game logic.